ksql-engine/src/main/java/io/confluent/ksql/function/udf/datetime/TimestampToString.java

```diff
@@ -28,18 +28,25 @@
 
   @Override
   public Object evaluate(final Object... args) {
-    if (args.length != 2) {
-      throw new KsqlFunctionException("TimestampToString udf should have two input argument:"
-                                      + " date value and format.");
+    if (args.length < 2) {
+      throw new KsqlFunctionException("TimestampToString udf should have at least "
+                                    + "two input arguments: date value and format.");
+    }
+
+    if (args.length > 3) {
+      throw new KsqlFunctionException("TimestampToString udf should have at most "
+                                    + "three input arguments: date value, format and zone.");
     }
 
     try {
       ensureInitialized(args);
 
       final Timestamp timestamp = new Timestamp((Long) args[0]);
+
       return timestamp.toLocalDateTime()
-          .atZone(ZoneId.systemDefault())
+          .atZone(args.length == 3 ? ZoneId.of(args[2].toString()) : ZoneId.systemDefault())
```

Comment:
I believe this just adds a time zone, without adjusting the value of the datetime itself. So if you have a timestamp of 1534722957, you'll get a date time of 08:19:2018 16:55:57 (assuming you're system default is PST). Then, if you call atZone(ZoneId.of("GMT")), you'll get back a zoned date time of 08:19:2018 16:55:57 GMT, which is not what we want. That's a different point in time than the original timestamp. I think what we want is to set the zone in the formatter. Also, can you add a test case that verifies that the zone adjustment is done correctly? Thanks!